src/main/scala/com/sksamuel/scapegoat/Inspection.scala

```diff
@@ -94,6 +97,18 @@ case class InspectionContext(global: Global, feedback: Feedback) {
         case _ => inspect(tree)
       }
     }
+
+    protected def isArray(tree: Tree): Boolean = tree.tpe.typeSymbol.fullName == "scala.Array"
+    protected def isTraversable(tree: Tree): Boolean = tree.tpe <:< typeOf[Traversable[Any]]
+    protected def isSeq(t: Tree): Boolean = t.tpe <:< typeOf[Seq[Any]]
+    protected def isIndexedSeq(t: Tree): Boolean = t.tpe <:< typeOf[IndexedSeq[Any]]
+    protected def isSet(t: Tree): Boolean = {
```

Comment:
Similar to the `Map` case discussed below, I think we need to do this method of searching base classes instead of `Set[Any]` because set is invariant in its type parameter.